test/e2e/e2ethanos/services.go

```diff
@@ -505,11 +505,33 @@ func (r *ReceiveBuilder) Init() e2e.InstrumentedRunnable {
 	}
 
 	if r.limit != 0 && r.metamonitoring != "" {
-		args["--receive.tenant-limits.max-head-series"] = fmt.Sprintf("%v", r.limit)
-		args["--receive.tenant-limits.meta-monitoring-url"] = r.metamonitoring
-		if r.metamonitoringQuery != "" {
-			args["--receive.tenant-limits.meta-monitoring-query"] = r.metamonitoringQuery
+		cfg := receive.RootLimitsConfig{
+			WriteLimits: receive.WriteLimitsConfig{
+				GlobalLimits: receive.GlobalLimitsConfig{
+					MetaMonitoringURL:        r.metamonitoring,
+					MetaMonitoringLimitQuery: r.metamonitoringQuery,
+				},
+				DefaultLimits: receive.DefaultLimitsConfig{
+					HeadSeriesLimit: uint64(r.limit),
+				},
+				// Can also add Per tenant limits.
```

Comment:
Do we want to merge this dead code?

Reply:
I was keeping this to be able to test types of config and behaviour fully, but can remove! 🙂